osidb/tests/test_flaw.py

```diff
@@ -970,6 +970,240 @@ def test_validate_major_incident_state(self, state, should_raise):
+            ),
+            # summary review missing
+            (
+                "mitigation text",
+                "statement text",
+                "summary text",
+                Flaw.FlawRequiresSummary.NOVALUE,
+                [
+                    FlawReference.FlawReferenceType.ARTICLE,
+                    "https://access.redhat.com/link123",
+                ],
+                True,
+                ["mi_summary_not_reviewed"],
+            ),
+            # summary review requested
+            (
+                "mitigation text",
+                "statement text",
+                "summary text",
+                Flaw.FlawRequiresSummary.REQUESTED,
+                [
+                    FlawReference.FlawReferenceType.ARTICLE,
+                    "https://access.redhat.com/link123",
+                ],
+                True,
+                ["mi_summary_not_reviewed"],
+            ),
+            # summary review not required
+            (
+                "mitigation text",
+                "statement text",
+                "summary text",
+                Flaw.FlawRequiresSummary.REJECTED,
+                [
+                    FlawReference.FlawReferenceType.ARTICLE,
+                    "https://access.redhat.com/link123",
+                ],
+                True,
+                ["mi_summary_not_reviewed"],
+            ),
+            # article missing
+            (
+                "mitigation text",
+                "statement text",
+                "summary text",
+                Flaw.FlawRequiresSummary.APPROVED,
+                [
+                    FlawReference.FlawReferenceType.EXTERNAL,
+                    "https://httpd.apache.org/link123",
+                ],
+                True,
+                ["mi_article_missing"],
+            ),
+        ],
+    )
+    def test_validate_major_incident_fields(
+        self,
+        mitigation,
+        statement,
+        summary,
+        requires_summary,
+        article,
+        should_alert,
+        alerts,
+    ):
+        """
+        Tests that a Flaw that is Major Incident complies with the following:
+        * has a mitigation
+        * has a statement
+        * has a summary
+        * requires_summary is APPROVED
+        * has exactly one article
+        """
+        flaw = FlawFactory.build(
+            major_incident_state=Flaw.FlawMajorIncident.APPROVED,
+            mitigation=mitigation,
+            statement=statement,
+            summary=summary,
+            requires_summary=requires_summary,
+            embargoed=False,  # to simplify fields that a flaw requires
+        )
+        flaw.save()
+
+        FlawReferenceFactory(flaw=flaw, type=article[0], url=article[1])
+        AffectFactory(flaw=flaw)
+        flaw.save()
+
+        if should_alert:
+            assert len(alerts) == len(flaw._alerts)
+            for alert in alerts:
+                assert alert in flaw._alerts
```

Comment:
As all test cases contain exactly one alert, I could change them to be of `str` type instead of `list` type, and then test directly `assert alert in flaw._alerts` without the `for` cycle.

Does anyone here agree with this and want me to change it? :D

Reply:
Makes sense to me but not insisting on it. If you want to improve it then great!